Sources/Implementation/DefaultDatafileHandler.swift

```diff
@@ -255,56 +257,57 @@ class DefaultDatafileHandler: OPTDatafileHandler {
         stopPeriodicUpdates()
     }
     
-    func saveDatafile(sdkKey: String, dataFile: Data) {
-        if let dir = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first {
-            
-            let fileURL = dir.appendingPathComponent(sdkKey, isDirectory: false)
-            
-            //writing
-            do {
-                try dataFile.write(to: fileURL, options: .atomic)
-            } catch {/* error handling here */
-                logger.e("Problem saving datafile for key " + sdkKey)
-            }
+    func getDatafileCache(sdkKey:String) -> OPTDataStore {
+        if let cache = datafileCache[sdkKey] {
+            return cache
+        }
+        else {
+            #if os(tvOS)
+            let store = DataStoreUserDefaults()
```

Comment:
It looks good, but I'm concerned about the datafile size which can grow big more than MB. They may not fit into UserDefaults max configs in many cases. Also with event queues, it can take substantial spaces in UserDefaults even if they can fit in 128KB each.

Reply:
even if you reach 512k in tvOS, the action is that the user is notified, it can actually grow beyond that.  So, this is probably pretty safe.  That said, if the datafile is too big, what choice do they have on tvOS?

Reply:
basically, the same thing will happen as happens now on tvOS.  start async, caching fails, but you can continue.  the bigger question may be do we want to disable polling for tvOS?  i am really curious why the simulator doesn't fail.

Reply:
Polling is optional, so we don't need to block it. They won't turn it on if not working. 
We can also consider "Cache" folder as done in ObjC. Both methods are not fully reliable for keeping datafile cache and events.